src/Kafka/Consumer.hs

```diff
@@ -1,8 +1,58 @@
 {-# LANGUAGE LambdaCase        #-}
 {-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE TupleSections     #-}
+
+-----------------------------------------------------------------------------
+-- |
+-- Module to consume messages from Kafka topics.
+-- 
+-- Here's an example of code to consume messages from a topic:
+-- 
+-- @
+-- import Control.Exception (bracket)
+-- import Data.Monoid ((<>))
+-- import Kafka
+-- import Kafka.Consumer
+-- 
+-- -- Global consumer properties
+-- consumerProps :: 'ConsumerProperties'
+-- consumerProps = 'brokersList' ['BrokerAddress' "localhost:9092"]
+--              <> 'groupId' ('ConsumerGroupId' "consumer_example_group")
+--              <> 'noAutoCommit'
+--              <> 'logLevel' 'KafkaLogInfo'
+-- 
+-- -- Subscription to topics
+-- consumerSub :: 'Subscription'
+-- consumerSub = 'topics' ['TopicName' "kafka-client-example-topic"]
+--            <> 'offsetReset' 'Earliest'
+-- 
+-- -- Running an example
+-- runConsumerExample :: IO ()
+-- runConsumerExample = do
+--     res <- bracket mkConsumer clConsumer runHandler
+--     print res
+--     where
+--       mkConsumer = 'newConsumer' consumerProps consumerSub
+--       clConsumer (Left err) = return (Left err)
+--       clConsumer (Right kc) = (maybe (Right ()) Left) \<$\> 'closeConsumer' kc
+--       runHandler (Left err) = return (Left err)
+--       runHandler (Right kc) = processMessages kc
+-- 
+-- -- Example polling 10 times before stopping
+-- processMessages :: 'KafkaConsumer' -> IO (Either 'KafkaError' ())
+-- processMessages kafka = do
+--     mapM_ (\_ -> do
+--                    msg1 <- 'pollMessage' kafka ('Timeout' 1000)
+--                    putStrLn $ "Message: " <> show msg1
+--                    err <- 'commitAllOffsets' 'OffsetCommit' kafka
+--                    putStrLn $ "Offsets: " <> maybe "Committed." show err
+--           ) [0 .. 10]
+--     return $ Right ()
+-- @
+-----------------------------------------------------------------------------
 module Kafka.Consumer
-( module X
+( KafkaConsumer
```

Comment:
Note I moved this up in the import list, as it was close to the bottom of the module, when I find it's one of the most important ones. I am totally fine reverting this change if you disagree